Panic when specified `--var-file` does not exist

If a `--var-file` passed in on the command line does not exists, nomad-pack panics with a nil pointer dereference runtime error.

```
 ➜  nomad-pack git:(main) ✗ stat config.vars.hcl
stat: cannot stat 'config.vars.hcl': No such file or directory

 ➜  nomad-pack git:(main) ✗ nomad-pack render -var-file=config.vars.hcl promtail      

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbbf748]

goroutine 1 [running]:
github.com/hashicorp/nomad-pack/internal/pkg/errors.HCLDiagsToWrappedUIContext(0xc00000e090, 0x1, 0x1, 0x1, 0x1, 0x2a)
        /home/circleci/project/project/internal/pkg/errors/wrapped_context.go:46 +0x1a8
github.com/hashicorp/nomad-pack/internal/pkg/manager.(*PackManager).ProcessTemplates(0xc00000c978, 0x18, 0x12f5920, 0x7ffe8e017d01, 0xc00000c978)
        /home/circleci/project/project/internal/pkg/manager/manager.go:82 +0x732
github.com/hashicorp/nomad-pack/cli.renderPack(0xc00000c978, 0x167e6e8, 0xc000553100, 0xc00091fca8, 0x0, 0x0, 0x2)
        /home/circleci/project/project/cli/helpers.go:235 +0x45
github.com/hashicorp/nomad-pack/cli.(*RenderCommand).Run(0xc00000c6f0, 0xc0000300a0, 0x2, 0x2, 0xc000119c80)
        /home/circleci/project/project/cli/render.go:55 +0x4c5
github.com/mitchellh/cli.(*CLI).Run(0xc0000ce140, 0xc0006bfef8, 0xc00000eb28, 0x1)
        /go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x41a
github.com/hashicorp/nomad-pack/cli.Main(0xc000030080, 0x4, 0x4, 0x0)
        /home/circleci/project/project/cli/main.go:94 +0x225
main.main()
        /home/circleci/project/project/main.go:12 +0xa5
```

--- a/cli/commands.go
+++ b/cli/commands.go
@@ -175,7 +175,6 @@ func (c *baseCommand) Init(opts ...Option) error {
 	return nil
 }
 
-
 func (c *baseCommand) ensureCache() error {
 	// Creates global cache
 	globalCache, err := cache.NewCache(&cache.CacheConfig{
@@ -187,7 +186,7 @@ func (c *baseCommand) ensureCache() error {
 	}
 
 	// Check if default registry exists
-	_, err = os.Stat(path.Join(cache.DefaultCachePath(),cache.DefaultRegistryName))
+	_, err = os.Stat(path.Join(cache.DefaultCachePath(), cache.DefaultRegistryName))
 	// If it does not error, then the registry already exists
 	if err == nil {
 		return nil
@@ -203,7 +202,6 @@ func (c *baseCommand) ensureCache() error {
 	}
 	return nil
 }
-
 
 // flagSet creates the flags for this command. The callback should be used
 // to configure the set with your own custom options.

--- a/cli/helpers.go
+++ b/cli/helpers.go
@@ -46,7 +46,6 @@ func initPackCommand(cfg *cache.PackConfig) *errors.UIErrorContext {
 	return errorContext
 }
 
-
 // TODO: Needs code review. This seems like it should get moved to the registry
 // package, in which case all of the dependent functions likely need to get moved.
 // get the global cache directory

--- a/internal/pkg/variable/parser.go
+++ b/internal/pkg/variable/parser.go
@@ -66,7 +66,7 @@ func NewParser(cfg *ParserConfig) (*Parser, error) {
 	for _, file := range cfg.FileOverrides {
 		_, err := os.Stat(file)
 		if err != nil {
-			return nil, errors.New(fmt.Sprintf("Variable file `%s` not found", file))
+			return nil, fmt.Errorf("variable file %q not found", file)
 		}
 	}
 

--- a/internal/pkg/variable/parser.go
+++ b/internal/pkg/variable/parser.go
@@ -3,6 +3,7 @@ package variable
 import (
 	"errors"
 	"fmt"
+	"os"
 	"sort"
 	"strings"
 
@@ -62,6 +63,12 @@ func NewParser(cfg *ParserConfig) (*Parser, error) {
 	// Sort the file overrides to ensure variable merging is consistent on
 	// multiple passes.
 	sort.Strings(cfg.FileOverrides)
+	for _, file := range cfg.FileOverrides {
+		_, err := os.Stat(file)
+		if err != nil {
+			return nil, errors.New(fmt.Sprintf("Variable file `%s` not found", file))
+		}
+	}
 
 	return &Parser{
 		fs: afero.Afero{

--- a/internal/runner/job/error.go
+++ b/internal/runner/job/error.go
@@ -50,8 +50,8 @@ func generateRegisterError(err error, errCtx *errors.UIErrorContext, jobName str
 
 	// Create our base error.
 	deployErr := errors.WrappedUIContext{
-		Err:      err,
-		Subject:  "failed to register job",
+		Err:     err,
+		Subject: "failed to register job",
 		Context: registerErr,
 	}
 

